### ISSUE TITLE
BOR Hazard Maps for two phase using wrong data

### ISSUE THREAD

**Issue by FLO-2DKaren** [FLO-2DSoftware/FLO-2DMapCrafter#66]
Hi Robson,

You need to change the data you  read for MapCrafter Hazard when looking at Two-Phase.  I think you should use the VxD or the Combined Max V * the Combined Max D.

I think they are only using the water map.

**Comment by FLO-2DKaren** [comment]
https://github.com/FLO-2DSoftware/FLO-2DMapCrafter/assets/20424460/c786e352-06a5-4a82-a89b-53c3656b2af6

**Comment by rpachaly** [comment]
Yes, we need to address these issues but the problem is not MapCrafter. 

USBR and SWISS methods:

We have the DEPTHMAX_2PHASE_COMBINED.OUT but we don't have a maximum velocity of both mudflow and water. We only have max velocity of water and max velocity of mudflow. That's why I did this considering only water.

**Comment by FLO-2DNoemi** [comment]
Combining the velocities is not straightforward. We can allow users to
select the velocities they want to use. A conservative approach would be to
use the mudflow velocities for this calculation, while a less conservative
approach would involve using the water velocities. We should ask Jim for
his input.

**Comment by FLO-2DNoemi** [comment]
Maybe a combination of the velocities at each cell. I think we could see
grids with water flow depths but mudflow velocity equal to zero, so if the
user selects the mudflow velocities for this calculation, the velocity will
be zero. I am not sure if I explain myself here.

**Comment by rpachaly** [comment]
In my opinion, it does not make sense to add the velocities or to make an average of it. I'm not sure if taking the max of the velocity or the mudflow for that specific grid element is also a good approach. 

MAYBE a weighted average makes sense, considering the depth and velocity of mudflow and water.

**Comment by FLO-2DNoemi** [comment]
I agree, it is not straightforward to combine these velocities, using the
water velocities for the product of H x V is the worst scenario.

**Comment by FLO-2DKaren** [comment]
What does the Vel_X_Depth.out show use?  Maybe we can extrapolate the velocity from that file?

**Comment by FLO-2DNoemi** [comment]
It reports the product of depth X Vel according to the following:

2 PHASE FLOW MAXIMUM VEL X DEPTH FOR FLUID AND MUDFLOW

VEL_X_DEPTH=0.
!.....MAXIMUM VEL_X_DEPTH...COMPARE WITH FLOODPLAIN VALUE

        IF(MUD.LE.1)THEN

 IF(VEL_X_DEPTH(J).LT.ABS(FCFD(I)*VELOC(I)))VEL_X_DEPTH(J)=ABS(FCFD(I)*VELOC(I))
        ENDIF

        IF(MUD.EQ.2)THEN

 IF(VEL_X_DEPTH(J).LT.(FCFD(I)*VELOC(I)))VEL_X_DEPTH(J)=(FCFD(I)*VELOC(I))


 IF(VEL_X_DEPTH_MUD(J).LT.ABS(FCFD_MUD(I)*VELOC_MUD(I)))VEL_X_DEPTH_MUD(J)=ABS(FCFD_MUD(I)*VELOC_MUD(I))

 IF(VEL_X_DEPTH_MUD(J).GT.VEL_X_DEPTH(J))VEL_X_DEPTH(J)=VEL_X_DEPTH_MUD(J)
        ENDIF

!......MAXIMUM VELOCITY X DEPTH

        No Mudflow
          IF(VEL(J,JDIR).LT.0.)THEN        Only when water velocity is less
than zero
          IF(VEL_X_DEPTH(J).LT.ABS(FPD(J)*VEL(J,JDIR)))THEN
           VEL_X_DEPTH(J)=ABS(FPD(J)*VEL(J,JDIR))
          ENDIF
          ENDIF

      2 PHASE FLOW MAXIMUM VEL X DEPTH FOR FLUID AND MUDFLOW
         IF(VEL(J,JDIR).LT.0.)THEN       Only when water velocity is less
than zero

IF(VEL_X_DEPTH(J).LT.ABS(FPD(J)*VEL(J,JDIR)))VEL_X_DEPTH(J)=ABS(FPD(J)*VEL(J,JDIR))
         ENDIF
         IF(VEL_MUD(J,JDIR).LT.0.)THEN   Only when mud velocity is less
than zero

IF(VEL_X_DEPTH_MUD(J).LT.ABS(FPD_MUD(J)*VEL_MUD(J,JDIR)))VEL_X_DEPTH_MUD(J)=ABS(FPD_MUD(J)*VEL_MUD(J,JDIR))

IF(VEL_X_DEPTH_MUD(J).GT.VEL_X_DEPTH(J))VEL_X_DEPTH(J)=VEL_X_DEPTH_MUD(J)

         ENDIF
